sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/ResourceManagerTestBase.java

```diff
@@ -319,6 +321,28 @@ private void setAccessible(final Field field) {
         });
     }
 
+    /**
+     * Sets http pipeline for internal use.
+     *
+     * @param httpPipeline the http pipeline
+     * @param azureConfigurable the azure configurable instance
+     * @param impl the class type of azure configurable implementation
+     * @param <AzureConfigurable> the type of azure configurable
+     * @param <AzureConfigurableImpl> the type of azure configurable implementation
+     * @return the azure configurable instance after setting internal http pipeline
+     * @throws RuntimeException when field cannot be found or set.
+     */
+    protected <AzureConfigurable, AzureConfigurableImpl> AzureConfigurable setInternalHttpPipeline(HttpPipeline httpPipeline, AzureConfigurable azureConfigurable, Class<AzureConfigurableImpl> impl) {
```

Comment:
An alternative might be export `com.azure.resourcemanager.resources.fluentcore.arm.models.implementation` to `azure-resourcemanager-test` as well.

Reply:
No, it's similar why we use a type parameter for sdk context as `azure-resourcemanager-test` is the base dependency of others.

Reply:
Ok, I see. Basically it just `T`, `S`.